pytest/lib/proxy.py

```diff
@@ -203,6 +203,18 @@ def check_finish(server, global_stopped, local_stopped, error):
         local_stopped.value = 2
 
 
+async def _read_exact(reader, length):
+    """Reads exactly `length` bytes from reader."""
+    data = b''
+    while True:
+        buf = await reader.read(length - len(data))
+        if not buf:
+            raise RuntimeError('Endpoint closed')
```

Comment:
So yeah, I think I don’t understand the code and perhaps this check should not be here? There’s bunch of NayDuck failures with this exception in logs.